src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Security/Permissions/PermissionProviderExtensions.cs

```diff
@@ -0,0 +1,23 @@
+namespace OrchardCore.Security.Permissions;
+
+public static class PermissionProviderExtensions
+{
+    public static async ValueTask<IEnumerable<Permission>> FindByNamesAsync(this IPermissionService permissionService, IEnumerable<string> names)
```

Comment:
Could be async enumerable and don't create an intermediate List then.

Reply:
We are making an asynchronous call. How can we return IEnumerable while calling await and not have intermediate list?

Reply:
`yield return`

https://learn.microsoft.com/en-us/archive/msdn-magazine/2019/november/csharp-iterating-with-async-enumerables-in-csharp-8#a-tour-through-async-enumerables

Reply:
That would force us to use IAsyncEnumerable the API won't be as simple. this is how I currently use it:
```
var permissions = await _permissionService.FindByNamesAsync(selectedPermissions);
```